javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -225,10 +218,10 @@ protected T getNext() {
                 @Override
                 public T next() { return delegate.previous(); }
             };
+        } else if (iterable instanceof Seq) {
+            return ((Seq<T>) iterable).reverseIterator();
         } else {
-            final Seq<? extends T> result = (iterable instanceof Seq) ? (Seq<T>) iterable
-                                                                      : Vector.ofAll(iterable);
-            return result.reverseIterator();
+            return Iterator.ofAll(iterable).foldLeft(List.<T>empty(), List::prepend).iterator();
```

Comment:
you're using a stack's iterator here (i.e. `List` is a `Stack`), which is basically a reverse iterator.
It's a better solution to the problem, than what I've written previously, i.e. create a `Vector` from it and traverse it backwards.